util/cache.go

```diff
@@ -0,0 +1,111 @@
+// Handle the password caching
+package util
+
+import (
+	"errors"
+	"time"
+
+	"github.com/patrickmn/go-cache"
```

Comment:
Is there a reason that you chose this key:value store in particular? It has a lot of open issues and pr's and it doesnt have a lot of current activity, so I dont feel super comfortable in adding it as a dependency to the project. Why not use something well known like bolt (https://github.com/coreos/bbolt) or another? There is a good list of these here https://github.com/avelino/awesome-go#resource-embedding

Reply:
Essentially, I don't want to store the decrypted user password anywhere on disk, so I was looking for a non-persistent key-value store which keeps things in memory. go-cache is thread-safe and handles things in memory. I have not yet used bbolt, so I am not sure, if we could use it  as a in-memory-only Store

Reply:
Ok, in-memory makes total sense for that use case. There are a few in memory suggestions here https://github.com/gostor/awesome-go-storage that maybe you could check out to see if they are better maintained.

Reply:
OK, from looking at it, we have BuntDB, which is also unmaintained, bigcache, which seems a bit oversized, https://github.com/bluele/gcache and  https://github.com/muesli/cache2go, which look like suitable replacements. I would go with cache2go, if you agree.

Reply:
cache2go looks good to me, thanks :)